### PR TITLE
fix: incorrect lot attributes parsing

### PR DESCRIPTION
**Pull request by jeffwheeler**
Previously, attributes were not parsed carefully, so dates and labels could not be extracted. Also, empty attributes (`{}`) would fail to parse. (Empty cost fields are valid when performing reductions, see [How Inventories Work](https://beancount.github.io/docs/how_inventories_work.html) in the Beancount documentation.)

This commit fixes the parsing, while preserving the existing API.

### CODE REVIEW

File: src/transaction/posting.rs [hunk 1]
```diff
@@ -153,7 +208,7 @@ mod tests {
         );
         assert_eq!(posting.amount(), Some(&Amount::new(10, "CHF")));
         assert!(posting.price().is_none());
-        assert!(posting.cost().is_none());
+        assert!(posting.lot_attributes().is_none());
```
Comment (jcornaz) [inline src/transaction/posting.rs:211]:
```suggestion
        assert!(posting.cost().is_none());
        assert!(posting.lot_attributes().is_none());
```

File: src/transaction/posting.rs [hunk 2]
```diff
@@ -93,6 +103,55 @@ pub enum PriceType {
     Total,
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub struct LotAttributes<'a> {
```
Comment (jcornaz) [inline src/transaction/posting.rs:107]: keep it private for now,

```suggestion
struct LotAttributes<'a> {
```

File: src/transaction/posting.rs [hunk 3]
```diff
@@ -69,10 +70,19 @@ impl<'a> Posting<'a> {
         self.info.price.as_ref().map(|(t, p)| (*t, p))
     }
 
+    /// Returns the lot attributes (if present)
+    #[must_use]
+    pub fn lot_attributes(&self) -> Option<&LotAttributes<'a>> {
```
Comment (jcornaz) [inline src/transaction/posting.rs:75]: keep it private for now

```suggestion
    fn lot_attributes(&self) -> Option<&LotAttributes<'a>> {
```

**Review by jcornaz — CHANGES_REQUESTED** [review]
Thank you very much!

As the tests are focused on fixing the bug in `Posting::cost`, I'd rather, keep `Posting::lot_attributes` private for now, and have this PR be a bug-fix only (no new features).

We may of course follow up with the `lot_attributes` feature, by making that member public and writing a couple of tests for it.

**Review by jcornaz — APPROVED** [review]
Thank you very much for the bug report and bug fix.

I made tiny adjustments (removed the new public members, and addressed clippy warnings) and will merge this PR, so that I can cut a patch release.

You may re-add the public `lot_attributes` in a follow-up PR if you want.

Thanks again.

**Comment by jeffwheeler** [comment]
Looks good, thanks for the follow-up commits 🙂

**Pull request merged**